editoast/src/views/test_app.rs

```diff
@@ -213,18 +224,45 @@ impl TestApp {
     pub fn get(&self, path: &str) -> TestRequest {
         self.server.get(&trim_path(path))
     }
+
     pub fn post(&self, path: &str) -> TestRequest {
         self.server.post(&trim_path(path))
     }
+
     pub fn put(&self, path: &str) -> TestRequest {
         self.server.put(&trim_path(path))
     }
+
     pub fn patch(&self, path: &str) -> TestRequest {
         self.server.patch(&trim_path(path))
     }
+
     pub fn delete(&self, path: &str) -> TestRequest {
         self.server.delete(&trim_path(path))
     }
+
+    pub async fn def_user(&self, user: &UserInfo, roles: HashSet<BuiltinRole>) {
+        let driver = PgAuthDriver::<BuiltinRole>::new(self.db_pool());
+        let uid = driver
+            .ensure_user(user)
+            .await
+            .expect("User should be created successfully");
+        driver
+            .ensure_subject_roles(uid, roles)
+            .await
+            .expect("Roles should be updated successfully");
+    }
+}
+
+pub trait WithUser {
+    fn with_user(self, user: UserInfo) -> Self;
```

Comment:
```suggestion
    fn from_user(self, user: UserInfo) -> Self;
```

A request comes from a user.

Reply:
I used this name, because I got the following error :
```bash
warning: methods called from_* usually take no self
   --> src/views/test_app.rs:258:18
    |
258 |     fn from_user(self, user: UserInfo) -> Self;
    |                  ^^^^
    |
    = help: consider choosing a less ambiguous name
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#wrong_self_convention
    = note: #[warn(clippy::wrong_self_convention)] on by default
```

Reply:
😑😑😑😑😑😑😑😑😑😑😑😑

Okay, so what about `by_user`, that should make clippy happy...